cmd/ovirt-populator/ovirt-populator.go

```diff
@@ -0,0 +1,252 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	populator_machinery "github.com/kubev2v/lib-volume-populator/populator-machinery"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	prefix     = "forklift.konveyor.io"
+	mountPath  = "/mnt/"
+	devicePath = "/dev/block"
+)
+
+var version = "unknown"
+
+const (
+	groupName  = "forklift.konveyor.io"
+	apiVersion = "v1beta1"
+	kind       = "OvirtVolumePopulator"
+	resource   = "ovirtvolumepopulators"
+)
+
+func main() {
+	var mode, engineUrl, secretName, diskID, fileName, crName, crNamespace, httpEndpoint, metricsPath, masterURL, kubeconfig, imageName, namespace string
+	var showVersion bool
+
+	// Main arg
+	flag.StringVar(&mode, "mode", "", "Mode to run in (controller, populate)")
+	// Populate args
+	flag.StringVar(&engineUrl, "engine-url", "", "ovirt-engine url (https//engine.fqdn)")
+	flag.StringVar(&secretName, "secret-name", "", "secret containing oVirt credentials")
+	flag.StringVar(&diskID, "disk-id", "", "ovirt-engine disk id")
+	flag.StringVar(&fileName, "file-name", "", "File name to populate")
+	flag.StringVar(&crName, "cr-name", "", "Custom Resource instance name")
+	flag.StringVar(&crNamespace, "cr-namespace", "", "Custom Resource instancce namespace")
+
+	// Controller args
+	if f := flag.Lookup("kubeconfig"); f != nil {
+		kubeconfig = f.Value.String()
+	} else {
+		flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	}
+	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flag.StringVar(&imageName, "image-name", "", "Image to use for populating")
+	// Metrics args
+	flag.StringVar(&httpEndpoint, "http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled.")
+	flag.StringVar(&metricsPath, "metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	// Other args
+	flag.BoolVar(&showVersion, "version", false, "display the version string")
+	flag.StringVar(&namespace, "namespace", "konveyor-forklift", "Namespace to deploy controller")
+	flag.Parse()
+
+	if showVersion {
+		fmt.Println(os.Args[0], version)
+		os.Exit(0)
+	}
```

Comment:
why do we need this?

Reply:
i guess if you wish to run the image as a standalone and just do version check. @bennyz fix me if i'm wrong?

Reply:
yeah it's copied from the example

Reply:
the example of a volume populator? unless I'm missing something - it is always be "unknown" here, right? I don't see an added value in having this..